test/integration_test.go

```diff
@@ -558,15 +561,39 @@ func (ts *IntegrationTestSuite) TestSignalWorkflow() {
 		ts.tracer.GetTrace("SignalWorkflow"))
 }
 
+func (ts *IntegrationTestSuite) TestSignalWorkflowWithInterceptorError() {
+	ctx, cancel := context.WithTimeout(context.Background(), ctxTimeout)
+	defer cancel()
+
+	// Return error 3 times from the interceptor
+	ts.inboundSignalInterceptor.returnErrorTimes = 3
+	wfOpts := ts.startWorkflowOptions("test-signal-workflow-interceptor-error")
+	run, err := ts.client.ExecuteWorkflow(ctx, wfOpts, ts.workflows.SignalWorkflow)
+	ts.Nil(err)
+	// Let workflow task run and send signal after to ensure correct order.
+	<-time.After(time.Second)
```

Comment:
Ideally let's get this a better way - don't want to introduce more flakes like the ones you're already fixing :)